crates/ide/src/doc_links.rs

```diff
@@ -483,10 +524,23 @@ fn get_doc_base_url(db: &RootDatabase, def: Definition) -> Option<Url> {
                     krate = name,
                     version = version.as_deref().unwrap_or("*")
                 ))
-            })?
+            });
+            (weblink, local_doc_path)
         }
     };
-    Url::parse(&base).ok()?.join(&format!("{display_name}/")).ok()
+    let web_base = web_base
+        .and_then(|it| Url::parse(&it).ok())
+        .and_then(|it| it.join(&format!("{display_name}/")).ok());
+    let local_base = local_base.and_then(|it| it.join(&format!("{display_name}/")).ok());
+
+    return (web_base, local_base);
+
+    fn create_url_from_os_str(path: &OsStr) -> Option<Url> {
+        let mut with_prefix = OsStr::new("file:///").to_os_string();
+        with_prefix.push(path);
+        with_prefix.push("/");
+        with_prefix.to_str().and_then(|s| Url::parse(s).ok())
+    }
```

Comment:
again, lets make this a closure and put it in front of its usage (or inline it directly)